docs/examples/userguide/convolve_memview.pyx

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+
+# "def" can type its arguments but not have a return type. The type of the
+# arguments for a "def" function is checked at run-time when entering the
+# function.
+# We now need to fix a datatype for our arrays. I've used the variable
+# DTYPE for this, which is assigned to the usual NumPy runtime
+# type info object.
+DTYPE = np.intc
+# The arrays f, g and h is typed as "np.ndarray" instances. The only effect
+# this has is to a) insert checks that the function arguments really are
+# NumPy arrays, and b) make some attribute access like f.shape[0] much
+# more efficient. (In this example this doesn't matter though.)
+def naive_convolve_memview(int [:,:] f, int [:,:] g):
+    if g.shape[0] % 2 != 1 or g.shape[1] % 2 != 1:
+        raise ValueError("Only odd dimensions on filter supported")
+    # The "cdef" keyword is also used within functions to type variables. It
+    # can only be used at the top indentation level (there are non-trivial
+    # problems with allowing them in other places, though we'd love to see
+    # good and thought out proposals for it).
+    #
+    # For the indices, the "int" type is used. This corresponds to a C int,
+    # other C types (like "unsigned int") could have been used instead.
+    # Purists could use "Py_ssize_t" which is the proper Python type for
+    # array indices.
+    cdef int vmax = f.shape[0]
+    cdef int wmax = f.shape[1]
+    cdef int smax = g.shape[0]
+    cdef int tmax = g.shape[1]
+    cdef int smid = smax // 2
+    cdef int tmid = tmax // 2
+    cdef int xmax = vmax + 2*smid
+    cdef int ymax = wmax + 2*tmid
+    h_np =  np.zeros([xmax, ymax], dtype=DTYPE)
+    cdef int [:,:] h = h_np
+    cdef int x, y, s, t, v, w

```

Comment:
I actually think that all the index/size variables should use `Py_ssize_t` rather than `int` as type, just to avoid providing bad examples.